oc/pod_exec.py

```diff
@@ -133,9 +140,26 @@ def run(self, podname, namespace, cmd_obj):
                 done = True
             else:
                 break
+        """
+        Couple of glitches in capturing return value.
+        Rest api doesn't return ret value of the command
+        hence this workaround.
+        we can fix this once we have facility to capture err code
+        """
         if cmd_obj.check_ec:
             resp.write_stdin("echo $?\n")
-            ret = int(resp.readline_stdout(timeout=5))
+            try:
+                ret = int(resp.readline_stdout(timeout=5))
+            except (ValueError, TypeError):
+                logger.error("TimeOut: Command timedout after {}".format(cmd_obj.timeout))
+                raise CommandFailed(" Failed to run \"{cmd}\"".
+                                    format(cmd=cmd_obj.cmd))
+
         resp.close()
+        if outbuf:
+            stdout = outbuf
 
         return (stdout, stderr, ret)
+
+class CommandFailed(Exception):
```

Comment:
Please create Exceptions.py in utils and move this func there

Reply:
Done. Moved exception to Exception.py inside oc directory. Once Vasu has run infra ready he will be moving exceptions to proper directory.